core/src/main/java/org/apache/carbondata/core/util/BlockletIndexUtil.java

```diff
@@ -193,7 +154,11 @@ private static BlockMetaInfo createBlockMetaInfo(
         CarbonFile carbonFile = FileFactory.getCarbonFile(carbonDataFile);
         return new BlockMetaInfo(new String[] { "localhost" }, carbonFile.getSize());
       default:
-        return fileNameToMetaInfoMapping.get(FileFactory.getFormattedPath(carbonDataFile));
+        if (!FileFactory.isFileExist(carbonDataFile)) {
```

Comment:
please add  comments with example how this will help, also add the percentage improvement of count(*) for partition and non partition table in the description

Reply:
Done.